test/libsolidity/smtCheckerTests/operators/ternary_operator_with_tuple.sol

```diff
@@ -0,0 +1,9 @@
+contract C {
+	function f() public pure {
+		true ? () : (); // bug fix test, proper handling of empty tuples
```

Comment:
Note that many things that are not supposed to return anything actually return an empty tuple. It would be best to add a few more tests to make sure they all work.

For example using a function with no return value causes this ICE too:
```solidity
function f() {}
function g() {}

contract C {
    function test() public {
        true ? f() : g();
    }
}
```

Another case are events and errors (both instances and types), though I can't think of a way those could be used in a ternary expression.

An attempt to return an empty tuple from a function triggers the ICE too:
```solidity
function f() {}
function g() {}

contract C {
    function test() public {
        return true ? f() : g();
    }
}
```

Reply:
Thanks for test ideas. Those 2 cases no longer cause ICE after the fix.
Where can I learn more about what returns an empty tuple? Is is somewhere in solidity docs?